LoopBack/LBModel.m

```diff
@@ -96,18 +96,38 @@ - (LBModel *)modelWithDictionary:(NSDictionary *)dictionary {
 
     [[model _overflow] addEntriesFromDictionary:dictionary];
 
-    [dictionary enumerateKeysAndObjectsUsingBlock:^(id key, id obj, BOOL *stop) {
+    for (NSString *key in dictionary) {
+        id obj = dictionary[key];
         SEL setter = NSSelectorForSetter(key);
 
+        // if obj is a NSNumber (including a boolean value) try to use a setter for the primitive type
+        if ([obj isKindOfClass:[NSNumber class]]) {
```

Comment:
Yes.  Since this enhancement is rather generic (not specific to codegen), I filed a separate PR #58 for review a few days back.  It includes unit tests for the new behavior.